src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs

```diff
@@ -332,20 +321,35 @@ public async Task PayoutAsync(Balance[] balances)
                 return;
             }
 
+            var latestBlockResp = await daemon.ExecuteCmdAllAsync<DaemonResponses.Block>(logger, EthCommands.GetBlockByNumber, new[] { (object) "latest", true });
+            var latestGasFee = latestBlockResp.FirstOrDefault(x => x.Error == null)?.Response.BaseFeePerGas;
             var txHashes = new List<string>();
 
             foreach(var balance in balances)
             {
                 try
                 {
+                    //Check if gas fee is below par range
+                    var lastPaymentDate = await cf.Run(con => paymentRepo.GetLastPaymentDateAsync(con, balance.PoolId, balance.Address));
+                    var maxGasLimit = lastPaymentDate.HasValue && (clock.UtcNow - lastPaymentDate.Value).TotalHours <= extraConfig.GasLimitToleranceHrs
+                        ? extraConfig.GasLimit
+                        : extraConfig.MaxGasLimit;
+                    if(latestGasFee > maxGasLimit)
```

Comment:
in the old version we had a check for latestGasFee.HasValue, do we need that here too?